src/main/java/gift/product/entity/Product.java

```diff
@@ -0,0 +1,89 @@
+package gift.product.entity;
+
+import gift.product.dto.request.ProductModifyRequest;
+import jakarta.persistence.*;
+
+import java.util.Objects;
+
+@Entity
+@Table(name = "products")
+public class Product {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, unique = true)
+    private Long giftId;
+
+    @Column(nullable = false)
+    private String giftName;
+
+    @Column(nullable = false)
+    private Integer giftPrice;
+
+    @Column(nullable = false)
+    private String giftPhotoUrl;
+
+    @Column(nullable = false)
+    private boolean isKakaoMDAccepted = true;
+
+    public Product(Long giftId, String giftName, Integer giftPrice, String giftPhotoUrl) {
+        this.giftId = giftId;
+        this.giftName = giftName;
+        this.giftPrice = giftPrice;
+        this.giftPhotoUrl = giftPhotoUrl;
+    }
+
+    public Product() {}
+
+    public Long getId() {
+        return id;
+    }
+
+    public Long getGiftId() {
+        return giftId;
+    }
+
+    public String getGiftName() {
+        return giftName;
+    }
+
+    public Integer getGiftPrice() {
+        return giftPrice;
+    }
+
+    public String getGiftPhotoUrl() {
+        return giftPhotoUrl;
+    }
+
+    public boolean getIsKakaoMDAccepted() {
+        return isKakaoMDAccepted;
+    }
+
+    public boolean isGiftNameValid(){
+        String pattern = "^[a-zA-Z0-9가-힣 ()\\[\\]\\+\\-\\&\\/\\_]*$";
+        return giftName.matches(pattern);
+    }
+
+    public void isKakaoMessageInclude(){
+        String censorshipWord = "카카오";
+        if(giftName.contains(censorshipWord)){
+            this.isKakaoMDAccepted = false;
+        }
+    }
+
+    public void modifyProduct(ProductModifyRequest productModifyRequest) {
+        if(productModifyRequest.giftId() != null && !Objects.equals(giftId, productModifyRequest.giftId())){
+            this.giftId = productModifyRequest.giftId();
+        }
```

Comment:
```suggestion
        var requestedGiftId = productModifyRequest.getId();
        if(requestedGiftId != null && !Objects.equals(giftId, requestedGiftId)){
            this.giftId = requestedGiftId;
        }
```

메소드를 너무 호출해서 코드도 길어지는 것 같은데 변수로 빼서 쓰는 건 어떨까요?